src/function.ts

```diff
@@ -28,46 +28,38 @@ export type AnyLambda = Function<any, any>;
  * })
  * ```
  */
-export class Function<P, O> {
+export class Function<P, O> implements IIntegration {
   readonly kind = "Function" as const;
 
   // @ts-ignore - this makes `F` easily available at compile time
   readonly __functionBrand: ConditionalFunction<P, O>;
 
-  constructor(readonly resource: aws_lambda.IFunction) {
-    return makeCallable(this, (call: CallExpr, context: VTL | ASL): any => {
```

Comment:
Looking at the code, it looks like `Function` is no longer callable.

Reply:
Ok, I now understand. The Function's interface is callable so TS won't complain. And we transform the code to the AST so the Function is never actually called - it is always called when we interpret the AST.

I assume this will change when we merge the closure serializer?

Reply:
yeah, in the closure serializer, I have a `native` integration, that is actually 3 functions right now: `bootstrap`, `prewarm`, `call`.

1. Bootstap - For wiring CDK
2. Prewarm - For anything that should run once for each function instance (clients)
3. Call - replaces the `func()` inline

May not stay this way, but the point is, each context can get whatever they need from the integrations in a well typed way.

Instead of "calling" func(), the context will do `new Integration(func).asl(call, context)` or `func.asl?.(call, context)`. The class just provides a nice wrapper for error handling for unsupported contexts.

Reply:
Re-reading your original statement, I should add the that class is never physically turned into an function. It is always an object that looks like a function. During interpretation of the AST, we wrap the object in `ref()` and then call the `.asl()`, `.vtl()`, etc properties on said object.

No more manipulating the object into a function with properties.

Reply:
Are you re-writing the AST for the closure serializer? How does `func()` translate into `bootstrap` -> `prewarm` -> `call`?

Reply:
yes. 

1. Compile
   1. find all integrations in each closure
   2. rewrite integration call to call `integration.native.call` inline in the closure
2. Synth
  1. Run bootstrap for each integration to wire up CDK
  2.  Add pre-warm to the "factory" portion of the closure supported by pulumi for each integration (code that runs before `handle`)
3. Runtime
   1. Pre-warm methods run
   2. Handler is run, eventually, calling the `integration.native.call`

Reply:
Love it!